src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/AWSSecretsManagerPlugin.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package com.mysql.cj.jdbc.ha.plugins;
+
+import com.mysql.cj.Messages;
+import com.mysql.cj.conf.ConnectionUrl;
+import com.mysql.cj.conf.PropertyKey;
+import com.mysql.cj.conf.PropertySet;
+import com.mysql.cj.log.Log;
+
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+
+public class AWSSecretsManagerPlugin implements IConnectionPlugin {
+  private final ICurrentConnectionProvider currentConnectionProvider;
+  private final IConnectionPlugin nextPlugin;
+  private final Log logger;
+  private final PropertySet propertySet;
+  private final ObjectMapper mapper = new ObjectMapper();
+  private final String region;
+  private final String secretId;
+
+  public AWSSecretsManagerPlugin(
+      ICurrentConnectionProvider currentConnectionProvider,
+      PropertySet propertySet,
+      IConnectionPlugin nextPlugin,
+      Log logger) {
+    this.currentConnectionProvider = currentConnectionProvider;
+    this.propertySet = propertySet;
+    this.nextPlugin = nextPlugin;
+    this.logger = logger;
+    this.region = propertySet.getStringProperty(PropertyKey.secretsManagerRegion).getValue();
+    this.secretId = propertySet.getStringProperty(PropertyKey.secretsManagerSecretId).getValue();
+  }
+
+  @Override
+  public void openInitialConnection(ConnectionUrl connectionUrl) throws SQLException {
+    AWSSecretsManager client  = AWSSecretsManagerClientBuilder.standard()
+        .withRegion(this.region)
+        .build();
+    GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest().withSecretId(this.secretId);
+    GetSecretValueResult getSecretValueResult;
+    String secret = "";
+    Properties updatedProperties = new Properties();
+    updatedProperties.putAll(connectionUrl.getOriginalProperties());
```

Comment:
I think we can update original properties directly.

Reply:
I don't think the ConnectionUrl class allows for editing/accessing the properties directly.